chromadb/api/segment.py

```diff
@@ -451,7 +451,7 @@ def _query(
             records = metadata_reader.get_metadata(ids=list(all_ids))
             metadata_by_id = {r["id"]: r["metadata"] for r in records}
             for id_list in ids:
-                metadata_list = [metadata_by_id[id] for id in id_list]
+                metadata_list = [metadata_by_id.get(id, None) for id in id_list]
```

Comment:
This solves a general issue in the distributed segment architecture. Which is data inconsistency issues between segments.

Reply:
Add a comment explaining this